dna/zomes/courses/code/src/course.rs

```diff
@@ -21,119 +21,223 @@ use std::convert::TryFrom;
-        Course {
+
+    // Constructor that creates a filled-in CourseData instance with modules
+    pub fn from(
+        title: String,
+        owner: Address,
+        course_anchor: Address,
+        timestamp: u64,
+        modules: Vec<Address>,
+    ) -> Self {
+        CourseData {
             title: title,
-            teacher_address: owner,
             modules: modules,
             timestamp: timestamp,
+            teacher_address: owner,
+            course_anchor: course_anchor,
+        }
+    }
+
+    // Copy constructor that saves us some typing effort during the update operations
+    pub fn from_instance(another_course_data: CourseData) -> Self {
+        CourseData {
+            title: another_course_data.title,
+            modules: another_course_data.modules,
+            timestamp: another_course_data.timestamp,
+            teacher_address: another_course_data.teacher_address,
+            course_anchor: another_course_data.course_anchor,
+        }
+    }
+
+    // Return this instance as a Holochain entry. Useful for all HDK actions
+    pub fn entry(&self) -> Entry {
+        Entry::App("course_data".into(), self.into())
+    }
+}
+
+impl CourseAnchor {
+    // Constructor
+    pub fn new(owner: Address, timestamp: u64) -> Self {
+        CourseAnchor {
+            timestamp: timestamp,
+            teacher_address: owner,
         }
     }
 
+    // Return this instance as a Holochain entry. Useful for all HDK actions
     pub fn entry(&self) -> Entry {
-        Entry::App("course".into(), self.into())
+        Entry::App("course_anchor".into(), self.into())
     }
 }
 
-////////////////////Course Entry Definition
-pub fn course_entry_def() -> ValidatingEntryType {
+/* *************** CourseAnchor HDK entry definition */
+pub fn course_anchor_def() -> ValidatingEntryType {
     entry!(
-        name: "course",
-        description: "this is the definition of course",
+        name: "course_anchor",
+        description: "Anchor entry that can be safely used to address a particular course",
         sharing: Sharing::Public,
         validation_package: || {
             hdk::ValidationPackageDefinition::Entry
         },
-        validation: | validation_data: hdk::EntryValidationData<Course>| {
+        validation: | validation_data: hdk::EntryValidationData<CourseAnchor>| {
             match validation_data{
                 EntryValidationData::Create { entry, validation_data } => {
-                    if !validation_data.sources().contains(&entry.teacher_address) {
-                        return Err(String::from("Only the teacher can create their courses"));
-                    }
-
-                    validate_course_title(&entry.title)
+                    validate_only_teacher_can_do(validation_data.sources(), &entry.teacher_address, "create")
                 },
                 EntryValidationData::Modify { new_entry, old_entry, validation_data, .. } => {
```

Comment:
Can the anchor ever be modified? I don't think so...

Reply:
Huh, you're right. That was just inertia in my thinking.